### ISSUE TITLE
Consider validation of '/system/archetype/os/name'

### ISSUE THREAD

**Issue by jrha** [quattor/template-library-core#178]
As mentioned in #1281 

For:
* We are relying upon it in components, so we should be able to control the contents.

Against:
* The broker will accept almost anything, it may be some time after the OS is added to the broker before the validation is hit at compile time, this could be painful.

**Comment by ned21** [comment]
Another potential challenge: t's used in different ways at different sites. So e.g. we use it for Linux/Solaris but in @jouvin's setup docs he uses it to differentiate between Linux distros.

**Comment by jrha** [comment]
Perhaps the validation should be nothing more than "is a complete lower-case word with no funny characters"

**Comment by jrha** [comment]
One of the types on #180 perhaps?

**Comment by jouvin** [comment]
I think that if we want to be able to test against it in other standard templates, it would be better to have an exhaustive list of possible values. Is it really much more than linux, centos, scientific-linux, rhel, debian, ubuntu? It looks to me as a rather finite list that we can update when needed...

**Comment by jrha** [comment]
I don't think we can meaningfully validate the content in any other way than enforcing a lowercase `string_trimmed`.
